### ISSUE TITLE
How to add BatchNorm layers in a DagNN?

### ISSUE THREAD

**Issue by lineshen199466** [vlfeat/matconvnet#1170]
Hi,
Can anyone help me with how to add BATCHNORM layers in DagNN? I'm confused with parameters setting.
Thank you.
heng.
---------------------------------------------------------------
classdef BatchNorm < dagnn.ElementWise
  properties
    numChannels
    epsilon = 1e-5
    opts = {'NoCuDNN'} % ours seems slightly faster
  end

  properties (Transient)
    moments
  end
........

And ...
my code :  net.addLayer('bnorm1', dagnn.BatchNorm(), {'conv1'}, {'bnorm1'}, {'bm1f','bm1b','bm1c'})
'conv1' has the shape - [16,16,64,64].
The Error occurs as "The MULTIPLIERS size does not match the DATA depth."

**Comment by sm176357** [comment]
`net.addLayer('bn1', dagnn.BatchNorm('numChannels',numChannels), ...
    {'x1'}, {'x2'}, ...
    {'bn_filter','bn_bias','bn_moments'});`
`net.params(net.getParamIndex('bn_filter')).value    = ones(numChannels,1,'single');`
`net.params(net.getParamIndex('bn_bias')).value    = zeros(numChannels,1,'single');`
`net.params(net.getParamIndex('bn_moments')).value = zeros(numChannels,2,'single');`

x1 - input to the batchnorm, x2 - output
numChannels should be 64 for you.
